### ISSUE TITLE
[a11y]: MultiSelect reading selected items somewhat broken

### ISSUE THREAD

**Issue by finken2** [carbon-design-system/carbon#19488]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.83.0

### React version

18.3.1

### Automated testing tool and ruleset

VoiceOver, Jaws, etc

### Assistive technology

VoiceOver, Jaws, etc

### Description

When using multiselect, you can specify the rendering of items via the structure of the items, or you can have a `itemToElement` function. There's also an `itemToString` function for providing a string representation of the items.

[This previous PR](https://github.com/carbon-design-system/carbon/pull/15633) added functionality to read the selected items when using a screenreader, but it only looks at `.text` of the item. It assumes all items will have a `.text` and doesn't do anything with `itemToString`. 

This issue was found during AVT for IBM Cloud Satellite on IBM Cloud.  I think they'd also like the "selected" or "unselected" as you go through the options, but that's probably a bigger issue. 

Secondary issue:
Storybook for Multiselect is really lacking. Most of the props aren't listed.  Maybe that's b/c many are the same as Dropdown, but it's not obvious. 

### WCAG 2.1 Violation

4.1.2, according to avt

### Reproduction/example

https://stackblitz.com/edit/lmalegv8?file=src%2FApp.jsx

### Steps to reproduce

Render a multiselect with selected items. Use a screen reader or look for the visually-hidden div that has the text for the selected items. You'll notice the named items are missing. Do the same with items that have a `.text` and the names are in there. 

![Image](https://github.com/user-attachments/assets/5f120b0a-e526-41d7-b8b9-d656b02a59fa)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by manuel-dev-ibm** [comment]
I'm facing a related issue with the `MultiSelect` component. If I provide a function to `itemToElement` that returns a `<div>` containing the elements I want to display in the dropdown, the items will appear blank. If I set the function to just return a string, it works fine but I'm not able to define rendering using HTML elements in the `itemToElement` prop.

![Image](https://github.com/user-attachments/assets/1b1f8a3d-ec58-4db7-93b3-2805744f0398)
